### PR TITLE
[RDY] Fix truecolor when in libvte and using tmux

### PR DESCRIPTION
**Pull request by DarkDefender**
This should fix colors not working when in tmux

### CODE REVIEW

File: src/nvim/tui/tui.c [hunk 1]
```diff
@@ -1575,7 +1576,7 @@ static void augment_terminfo(TUIData *data, const char *term,
   // can use colons like ISO 8613-6:1994/ITU T.416:1993 says.
   bool has_colon_rgb = false
     // per GNOME bug #685759 and bug #704449
-    || (vte_version >= 3600)
+    || (vte_version >= 3600 && !tmux && !screen)
     || iterm || iterm_pretending_xterm  // per analysis of VT100Terminal.m
     // per http://invisible-island.net/xterm/xterm.log.html#xterm_282
     || true_xterm;
```
Comment (justinmk) [inline src/nvim/tui/tui.c]: Why only for vte? Shouldn't it be:

```c
bool has_colon_rgb = !tmux && !screen
  && ((vte_version >= 3600)
      || iterm || iterm_pretending_xterm
      || true_xterm);
```

Comment (justinmk) [inline src/nvim/tui/tui.c]: Looking at https://github.com/neovim/neovim/commit/8898793adeb3a82fe50da4258c30940e10ebcc9d#diff-3ac88981a139ac0eee448a6bd19c7569L1539 it seems like `!old_truecolor_env && !rxvt && ...` are also worth including as positive knowledge that `has_colon_rgb=false`.

Comment (DarkDefender) [inline src/nvim/tui/tui.c]: The check is only for vte because the others depent on what `TERM` is set to.
So you will not need to check if we are for example rxvt because then we can be sure that it didn't detect iterm in that case and so on.

Comment (justinmk) [inline src/nvim/tui/tui.c]: Ok, but that sounds like the logic is currently correct but it depends on the current configuration of previous checks. Why not make it explicit?

Regarding at least my first comment above: irrespective of `TERM`, _if_ we detect tmux/screen, then (correct me if wrong), _nothing else matter_. It doesn't matter if we also detected `iterm` or `iterm_pretending_xterm`; while running in tmux we _must not_ emit colon codes.

File: src/nvim/tui/tui.c [hunk 2]
```diff
@@ -1537,11 +1537,11 @@ static void augment_terminfo(TUIData *data, const char *term,
   bool teraterm = terminfo_is_term_family(term, "teraterm");
   bool putty = terminfo_is_term_family(term, "putty");
   bool screen = terminfo_is_term_family(term, "screen");
+  bool tmux = terminfo_is_term_family(term, "tmux") || !!os_getenv("TMUX");
   bool iterm = terminfo_is_term_family(term, "iterm")
     || terminfo_is_term_family(term, "iTerm.app");
   // None of the following work over SSH; see :help TERM .
   bool iterm_pretending_xterm = xterm && iterm_env;
-  bool tmux_wrap = screen && !!os_getenv("TMUX");
```
Comment (justinmk) [inline src/nvim/tui/tui.c]: why was this removed? AFAIK `TMUX_WRAP` should not be used in the real `screen`.

Comment (DarkDefender) [inline src/nvim/tui/tui.c]: That check was for it tmux had `TERM` set to screen. Because we moved the getenv to tmux we now know if it is tmux regardless what term is set to

**Comment by iredelmeier** [comment]
This fixes #7041 for me.

File: src/nvim/tui/tui.c (hunk 2 above)
Comment (justinmk) [inline src/nvim/tui/tui.c]: Just taking the old logic at it's word:

    bool tmux_wrap = screen && !!os_getenv("TMUX");
 
that specifically says "we only want to use `TMUX_WRAP` in the specific case where `TERM=screen-256color` and `$TMUX` is defined". That leads me to believe that we should not use `TMUX_WRAP` if `TERM=tmux-256color`. Was the old logic wrong?

Comment (DarkDefender) [inline src/nvim/tui/tui.c]: I think that the old logic was wrong, yes.

**Comment by DarkDefender** [comment]
There doesn't seem to be any obvious issues that should be corrected in the PR, so I'll mark it as RDY

**Review by mhinz — APPROVED** [review]
LGTM. This fixes tmux detection for me, since I use `tmux-256color` and not `screen-256color`, which is the recommendation of the tmux project.

**Comment by justinmk** [comment]
> This fixes tmux detection for me, since I use tmux-256color and not screen-256color, which is the recommendation of the tmux project.

@mhinz do you mean it still doesn't work with `screen-256color` ?

**Comment by justinmk** [comment]
Merged in https://github.com/neovim/neovim/pull/7100. Thanks a lot @DarkDefender .

Would like to discuss more with @jdebp but we need to get 0.2.1 in a working state very soon.
